### ISSUE TITLE
Pausing doesn't work

### ISSUE THREAD

**Issue by sbruder** [raatmarien/red-moon#10]
Pausing the filter doesn't work with version 2.2.1. It only stopps the the filter and hides the notifcation (so I must go in the app again and start it there).

**Comment by raatmarien** [comment]
Thanks for letting me know!

I'm not able to replicate this on my device with v2.2.1. Could you tell me what Android version you are running?

**Comment by sbruder** [comment]
I am using Android 4.4.4 (CM11) and the Red Moon version from F-Droid

**Comment by raatmarien** [comment]
Okay, thanks. I will look into this with an emulator soon.

**Comment by sbruder** [comment]
I've tested it just now with an device running android 5.0 and it appears there, too.

**Comment by raatmarien** [comment]
Sorry, I tested it on my phone and on emulators, but wasn't able to replicate the problem.

If you have the time, it would help if you could give me precise instructions as to how to replicate the bug. However, the way the pause state is handled will be different in the next release, so the bug might not persist in the next version. Taking that into account it might be more time effective to wait and see if the problem still persists in the next version.

**Comment by sbruder** [comment]
Ok, I'll wait for the next release and test it then.

**Comment by danialbehzadi** [comment]
Same for me on CM11 without gapps

**Comment by sbruder** [comment]
I've built the app just now, and there is the same problem again.
You can reproduce it, on a device with CM11 (Android 4.4.4) without GApps. Simply start it without any changes to the settings and then start it. Then swipe the notification bar down and press on pause. One second you can see the notification, but then it is not visible. In the app itself the slider is grey and on the left.
I hope you can reproduce the bug.

**Comment by raatmarien** [comment]
Thanks for the further testing @sbruder!

I've been looking into it and I think I found the bug. When the Fragment was running and the pause button in the notification was pressed, the Fragment would attempt to update the switch in the action bar to reflect the state change. However, the switch would then react to being changed to the 'unchecked' state and in turn turn off the filter. This caused the notification to disappear.

I wasn't able to replicate this issue before, because the Fragment was garbage collected on my device, before I could use the notification.

I think the bug is now fixed, but let me know if you still experience it after an update. Thanks for the help with debugging!
